x-pack/plugins/ml/public/register_helper/register_search_links/search_deep_links.ts

```diff
@@ -8,224 +8,193 @@
 import { i18n } from '@kbn/i18n';
 import type { LinkId } from '@kbn/deeplinks-ml';
 
-import { type AppDeepLink, AppNavLinkStatus } from '@kbn/core/public';
+import { type AppDeepLink } from '@kbn/core/public';
 import { ML_PAGES } from '../../../common/constants/locator';
 import type { MlCapabilities } from '../../shared';
 
-function createDeepLinks(
-  mlCapabilities: MlCapabilities,
-  isFullLicense: boolean,
-  showMLNavMenu: boolean
-) {
-  function getNavStatus(
-    visible: boolean,
-    showInServerless: boolean = true
-  ): AppNavLinkStatus | undefined {
-    if (showMLNavMenu === false) {
-      // in serverless the status needs to be "visible" rather than "default"
-      // for the links to appear in the nav menu.
-      return showInServerless && visible ? AppNavLinkStatus.visible : AppNavLinkStatus.hidden;
-    }
-
-    return visible ? AppNavLinkStatus.default : AppNavLinkStatus.hidden;
```

Comment:
In this file, instead of relying on the `hidden` status to remove a link based on app capabilities, I removed the deepLink entirely